File: python/rotate_array.py
# no. 189
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        optimize_step = k % len(nums)
        split_point = len(nums) - optimize_step
        new_nums = nums[split_point:] + nums[:split_point]
        for i in range(len(nums)):
           nums[i] = new_nums[i]

File: C++/rotate_array.cpp
// no. 189
#include <vector>
using namespace std;
class Solution {
public:
  void rotate(vector<int> &nums, int k) {
    int optimize_step = k % nums.size();
    int split_point = nums.size() - optimize_step;
    vector<int> new_nums;
    for (int i = 0; i < nums.size(); i++) {
      if (i + split_point < nums.size()) {
        new_nums.push_back(nums[i + split_point]);
      } else {
        new_nums.push_back(nums[i + split_point - nums.size()]);
      }
    }
    for (int i = 0; i < nums.size(); i++) {
      nums[i] = new_nums[i];
    }
  }
};
